modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/js/ckeditor5/plugins/AiCreator.tsx

```diff
@@ -0,0 +1,194 @@
+			popover.setAttribute('role', 'alert');
+			popover.setAttribute('tabindex', '0');
+
+			document.body.appendChild(popover);
+
+			requestAnimationFrame(() => {
+				const buttonRect = aiCreatorButton!.getBoundingClientRect();
+				const popoverRect = popover!.getBoundingClientRect();
+
+				if (popover) {
+					popover.style.top = `${buttonRect.top - popoverRect.height}px`;
+					popover.style.left = `${Math.floor(
+						buttonRect.left +
+							buttonRect.width / 2 -
+							popoverRect.width / 2
+					)}px`;
+				}
+			});
+		}
+
+		const config: Config<LiferayEditorConfig> = editor.config;
+
+		const aiCreatorOpenAIURL = config.get('aiCreatorOpenAIURL') as string;
+		const aiCreatorPortletNamespace = config.get(
+			'aiCreatorPortletNamespace'
+		);
+		const contentsLanguage = config.get('contentsLanguage') as string;
+		const isAICreatorOpenAIAPIKey = config.get('isAICreatorOpenAIAPIKey');
+
+		editor.commands.add('openAICreatorDialog', new Command(editor));
+		editor.commands.add(
+			'openAICreatorConfigurationPopover',
+			new Command(editor)
+		);
+
+		const openAICreatorDialogCommand = editor.commands.get(
+			'openAICreatorDialog'
+		)!;
+		const openAICreatorConfigurationPopoverCommand = editor.commands.get(
+			'openAICreatorConfigurationPopover'
+		)!;
+
+		openAICreatorDialogCommand.execute = () => {
+			const closeModalHandler = Liferay.on('closeModal', (event) => {
+				closeModalHandler.detach();
+
+				if (event.text) {
+					editor.model.change((writer) => {
+						let insertPosition =
+							editor.model.document.selection.getFirstPosition();
+
+						if (!insertPosition) {
+							const root = editor.model.document.getRoot();
+							if (!root) {
+								return;
+							}
+							insertPosition = writer.createPositionAt(
+								root,
+								'end'
+							);
+						}
+
+						writer.insertText(event.text, insertPosition);
+					});
+				}
+			});
+
+			const url = new URL(aiCreatorOpenAIURL);
+
+			url.searchParams.set(
+				`${aiCreatorPortletNamespace}languageId`,
+				contentsLanguage
+			);
+
+			Liferay.Util.openModal({
+				height: '550px',
+				onClose: () => closeModalHandler.detach(),
+				size: 'lg',
+				title: Liferay.Language.get('ai-creator'),
+				url: url.toString(),
+			});
+		};
+
+		openAICreatorConfigurationPopoverCommand.execute = () => {
+			showPopover();
+
+			const removePopover = () => {
+				popover?.removeEventListener('blur', removePopover);
+				hidePopover();
+			};
+
+			requestAnimationFrame(() => {
+				popover?.focus();
+				popover?.addEventListener('blur', removePopover);
+			});
+		};
+
+		editor.ui.componentFactory.add('aiCreator', () => {
+			const buttonView = new ButtonView();
+
+			buttonView.set({
+				icon: aiCreatorIcon,
+				label: Liferay.Language.get('ai-creator'),
+				tooltip: Liferay.Language.get('create-ai-content'),
+				withText: true,
+			});
+
+			buttonView
+				.bind('isEnabled')
+				.to(openAICreatorDialogCommand, 'isEnabled');
```

Comment:
To test if this binding works correctly, you can open source editing. If all is good, icon should be disabled.

Reply:
Oh yeah, there is already a PW test for this, so we would see this not working. All good :+1: